client/network/src/config.rs

```diff
@@ -643,6 +731,88 @@ impl NonReservedPeerMode {
 	}
 }
 
+///Doc
+#[derive(Clone, Debug)]
+pub enum PreShareKeyConfig {
+	/// A pre-shared key configuration.
+	PRESHAREDKEY(PreSharedKeySecret),
+}
+
+/// The options for obtaining a pre-shared key.
+pub type PSKey = PreSharedKeySecret;
+
+///Doc
+#[derive(Clone)]
+pub enum PreSharedKeySecret {
+	/// Read the pre-shared key from a file.
+	File(PathBuf),
+}
+
+impl Default for PreShareKeyConfig {
+	fn default() -> PreShareKeyConfig {
+		let path = PathBuf::from("not-used");
+		let key_bytes: [u8;32] = [24, 97, 125, 255, 78, 254, 242, 4, 80, 221, 94, 175, 192, 96, 253,
+		133, 250, 172, 202, 19, 217, 90, 206, 59, 218, 11, 227, 46, 70, 148, 252, 215];
+		fs::write(&path, hex::encode(key_bytes.as_ref())).expect("Writes pre shared key");
+		PreShareKeyConfig::PRESHAREDKEY(PreSharedKeySecret::File(path))
+	}
+}
+
+impl fmt::Debug for PreSharedKeySecret {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match self {
+			Self::File(path) => f.debug_tuple("PSKSecret::File").field(path).finish(),
+		}
+	}
+}
+
+impl PreShareKeyConfig {
+	/// Evaluate a `PreSharedKeyConfig` to obtain an identity `pre shared key`:
+	///
+	///  * If the secret is configured as a file, it is read from that file, if it exists. Otherwise
+	///    return error.
+	pub fn into_pre_share_key(self) -> io::Result<PreSharedKey> {
+		use PreShareKeyConfig::*;
+		match self {
+			PRESHAREDKEY(PreSharedKeySecret::File(f)) => {
+				match std::fs::read(f) {
+					Ok(_data) => {
+						match _data.len() {
+							32 => {
```

Comment:
let's assume we will keep this key in hex - then we can simplify and remove this variant

Reply:
ok